pkgs/applications/science/math/giac/default.nix

```diff
@@ -9,18 +9,28 @@ assert (!blas.isILP64) && (!lapack.isILP64);
 
 stdenv.mkDerivation rec {
   pname = "giac${lib.optionalString enableGUI "-with-xcas"}";
-  version = "1.9.0-43"; # TODO try to remove preCheck phase on upgrade
+  version = "1.9.0-993";
 
   src = fetchurl {
     url = "https://www-fourier.ujf-grenoble.fr/~parisse/debian/dists/stable/main/source/giac_${version}.tar.gz";
-    sha256 = "sha256-466jB8ZRqHkU5XCY+j0Fh7Dq/mMaOu10rHECKbtNGrs=";
+    sha256 = "sha256-pqytFWrSWfEwQqRdRbaigGCq68s8mdgj2j8M+kclslE=";
   };
 
   patches = [
+    ./remove-old-functional-patterns.patch
```

Comment:
Seems like your patch was upstreamed in the meantime! Thank you.

https://github.com/geogebra/giac/commit/e1f2e8899f7a579168f5565a05adf987457ddcff
https://github.com/geogebra/giac/commit/2f6ea6cd85ccc224d5ce2f8947ecbd2cc7840f1b